source/dredd-hooks.js

```diff
@@ -1,8 +1,29 @@
-var hooks = require('hooks');
+var hooks = require('hooks')
+  , typer = require('media-typer')
+  , yaml = require('js-yaml');
 
 hooks.beforeEachValidation(function (transaction) {
-  // Remove the trailing (or well, empty) new line
-  if (transaction.expected.headers['Content-Type'] === 'text/vnd.apiblueprint') {
-    transaction.real.body = transaction.expected.body.replace(/^$/g, '');
+  var type = typer.parse(transaction.expected.headers['Content-Type']);
+
+  // Indent the request and response bodies properly
+  if (type.suffix === 'json' || type.subtype === 'json') {
+    transaction.expected.body = JSON.stringify(JSON.parse(transaction.expected.body));
+    transaction.real.body = JSON.stringify(JSON.parse(transaction.real.body));
+
+    // Change content type because otherwise gavel would compare JSON objects
```

Comment:
Wouldn't this mean that we would no longer validate the real content-type is correct? We only checked that it was JSON before, now we are setting it to `text/plain` without verifying that the full expected content-type matches the real content type.

This hook would miss implementation problems such as if the server sent the wrong content type such as if there was a typo or a different version.

If we're going to set to `text/plain`, please validate the content-types match beforehand.